libs/server/Storage/Session/ObjectStore/SortedSetOps.cs

```diff
@@ -604,6 +604,69 @@ public unsafe GarnetStatus SortedSetDifference(ArgSlice[] keys, out Dictionary<b
             return GarnetStatus.OK;
         }
 
+        /// <summary>
+        /// Computes the difference between the first and all successive sorted sets and store resulting pairs in the destination key.
+        /// </summary>
+        /// <param name="keys"></param>
+        /// <param name="destinationKey"></param>
+        /// <param name="count"></param>
+        /// <returns></returns>
+        public unsafe GarnetStatus SortedSetDifferenceStore(ReadOnlySpan<ArgSlice> keys, ArgSlice destinationKey, out int count)
+        {
+            count = default;
+
+            if (keys.Length == 0)
+                return GarnetStatus.OK;
+
+            var createTransaction = false;
+
+            if (txnManager.state != TxnState.Running)
+            {
+                Debug.Assert(txnManager.state == TxnState.None);
+                createTransaction = true;
+                txnManager.SaveKeyEntryToLock(destinationKey, true, LockType.Exclusive);
+                foreach (var item in keys)
+                    txnManager.SaveKeyEntryToLock(item, true, LockType.Shared);
+                _ = txnManager.Run(true);
+            }
+
+            var setObjectStoreLockableContext = txnManager.ObjectStoreLockableContext;
+
+            try
+            {
+                var status = SortedSetDifference(keys, out var result);
```

Comment:
Note that SortedSetDifference is also doing locking (that you don't need at that point), there should be a common method that extracts only the common logic that both SSD & SSDS call. See SetDiff & SetDiffStore for reference.

Reply:
Fixed